### DISCUSSION TITLE
Allow immediate execution of lambdas with implicit return type

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/csharplang#7728] [Language Ideas] [answered]
Currently immediate execution of lambda with implicit return type throws `CS0149: Method name expected`. This seems like it should work considering there's no error if I explicitly specify the return type or assign the lambda to a variable first:

```cs
List<Task> tasks = [];
string s;

tasks.Add((async () => s = await GetStringAsync())()); // CS0149: Method name expected

tasks.Add(((Func<Task>)(async () => s = await GetStringAsync()))()); // OK

var getTask = async () => s = await GetStringAsync();
tasks.Add(getTask()); // OK

Task<string> GetStringAsync() => throw new NotImplementedException();
```

Motivation is: I want to add task to a list witch will be awaited with `Task.WhenAll` later on, and assign variable `s` with it's result within the same task. Wrapping `async () => s = await GetStringAsync()` in `Task.Run` would work as well, but it would create an unnecessary thread

**Comment by HaloFour** [comment, marked as answer]
See: #4748

Which was considered ["universally disliked"](https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-09-28.md#ungrouped) by the language design team.

**Reply by maxkoshevoi** [reply]
I see, thank you. Well we still need some concise syntax to do this. Hope expression blocks will be implemented soon =)

**Reply by yaakov-h** [reply]
"need" is a strong word here.
